.gitlab/dev_container_deploy/docker_linux.yml

```diff
@@ -17,7 +17,17 @@ dev_branch-dogstatsd:
 dev_branch_multiarch-a7:
   extends: .docker_publish_job_definition
   stage: dev_container_deploy
-  rules: !reference [.manual]
+  rules:
+    - !reference [.except_mergequeue]
+    - if: '$CI_COMMIT_BRANCH == "main"'
+      when: manual
+      variables:
+        IMAGE_TAG: "${CI_COMMIT_SHORT_SHA}"
+      allow_failure: true
+    - when: manual
+      variables:
+        IMAGE_TAG: "${CI_COMMIT_REF_SLUG}"
+      allow_failure: true
```

Comment:
If we are to modify the other job, I would keep both tags and run the job in parallel so we have both `master-py3` and `<hash>-py3`